extensions/positron-reticulate/src/extension.ts

```diff
@@ -107,9 +105,77 @@ export class ReticulateRuntimeManager implements positron.LanguageRuntimeManager
 	}
 
 	async createSession(runtimeMetadata: positron.LanguageRuntimeMetadata, sessionMetadata: positron.RuntimeSessionMetadata): Promise<positron.LanguageRuntimeSession> {
+		const sessionPromise = new PromiseHandles<positron.LanguageRuntimeSession>();
+		vscode.window.withProgress({
+			location: vscode.ProgressLocation.Notification,
+			title: 'Creating the Reticulate Python session',
+			cancellable: false
+		}, async (progress, _token) => {
+			let session: ReticulateRuntimeSession | undefined;
+			try {
+				session = await this.createSession_(runtimeMetadata, sessionMetadata, progress);
+				sessionPromise.resolve(session);
+			} catch (err) {
+				sessionPromise.reject(err);
+			}
+			// Wait for the session to start (or fail to start) before
+			// returning from this callback, so that the progress bar stays up
+			// while we wait.
+			if (session) {
+				// Wait for the session to start (or fail to start) before
+				// returning from this callback, so that the progress bar stays up
+				// while we wait.
+				progress.report({ increment: 10, message: 'Waiting to connect' });
```

Comment:
Nit: String should be localized (here and elsewhere)

Reply:
Done in https://github.com/posit-dev/positron/pull/7024/commits/d0e0e5937d3b5db369af4e87b06f2809801886e1